src/librustdoc/clean/blanket_impl.rs

```diff
@@ -14,10 +14,9 @@ crate struct BlanketImplFinder<'a, 'tcx> {
 }
 
 impl<'a, 'tcx> BlanketImplFinder<'a, 'tcx> {
-    // FIXME(eddyb) figure out a better way to pass information about
-    // parametrization of `ty` than `param_env_def_id`.
-    crate fn get_blanket_impls(&mut self, ty: Ty<'tcx>, param_env_def_id: DefId) -> Vec<Item> {
-        let param_env = self.cx.tcx.param_env(param_env_def_id);
+    crate fn get_blanket_impls(&mut self, item_def_id: DefId) -> Vec<Item> {
+        let param_env = self.cx.tcx.param_env(item_def_id);
+        let ty = self.cx.tcx.type_of(item_def_id);
```

Comment:
I'm guessing the addition of these `type_of` calls shouldn't cause a performance regression because it's a query?

Reply:
I wouldn't expect it to, no. The slow part of this function is looking at all the impls anyway, this isn't called in a loop.